classes/install/Upgrade.inc.php

```diff
@@ -1903,6 +1903,53 @@ function moveReviewerFiles() {
 			$result->MoveNext();
 		}
 		$result->Close();
+
+	}
+
+	/**
+	 * Move 'date_published' from the table 'published_submissions' into the table 'submission_settings'
+	 * @return boolean
+	 */
+	function migrateDatesPublished() {
+		$submissionDao = Application::getSubmissionDAO();
+		$result = $submissionDao->retrieve('SELECT submission_id, date_published FROM published_submissions');
+
+		while (!$result->EOF) {
+			$row = $result->GetRowAssoc(false);
+			$submissionId = $row['submission_id'];
+			$datePublished = $submissionDao->datetimeFromDB($row['date_published']);
+
+			$submission = $submissionDao->getById($submissionId);
+			$submission->setDatePublished($datePublished);
+			$submissionDao->updateLocaleFields($submission);
```

Comment:
Maybe just to use the SQL insert statement instead of getting the submission objects and update them -- the upgrade scripts work with SQL statements, so it would be OK here to do so, I suppose... Maybe just only one SQL statement for everything (for the whole function) would also work? -- In that case we could put it in an XML file in the upgrade folder...